src/components/TodoItem.tsx

```diff
@@ -0,0 +1,65 @@
+import React, {useState} from 'react';
+import {ITask} from '../models/ITask';
+import {useDispatch} from 'react-redux';
+import {completeTodos, removeTodos, updateTodos} from '../store/slices/todo.slice';
+
+interface TodoItemProps {
+    todoItem: ITask
+};
+
+export const TodoItem = ({todoItem}: TodoItemProps) => {
+    const [isEdit, setIsEdit] = useState(false);
+    const [task, setTask] = useState<ITask>(todoItem);
```

Comment:
I think this is an example of a component that "knows too much".
Having `const [task, setTask] = useState<ITask>(todoItem);` in this component leads to a bug:
- click edit on a todo item
- change the text
- click cancel
- click edit again
- the user expects to see the current label inside the input box, but the text from the last edit is displayed instead

The bug wouldn't happen if you had extracted an `EditTodoItem` component and put the line
```const [task, setTask] = useState<ITask>(todoItem);```
in there, because the component would be unmounted when `isEdit` is false, and the state would be reset.